File: app/back/src/main/java/net/seijishikin/jp/normalize/shuushi_doc/v05/constants/ShuushiDocV5KatsudouKuikiConstants.java
package net.seijishikin.jp.normalize.shuushi_doc.v05.constants;

/**
 * 政治資金収支報告書V5活動区域定数
 */
public final class ShuushiDocV5KatsudouKuikiConstants {

    /** 2以上の都道府県の区域等 */
    public static final short COMPLEX_PREF = 1;

    /** 同一の都道府県の区域内 */
    public static final short DOUITSU = 2;

    /** 2以上の都道府県の区域等 */
    public static final String COMPLEX_PREF_TEXT = "2以上の都道府県の区域等";

    /** 同一の都道府県の区域内 */
    public static final String DOUITSU_TEXT = "同一の都道府県の区域内";

    /**
     * コンストラクタ
     */
    private ShuushiDocV5KatsudouKuikiConstants() {
    }

    /**
     * 表示ラベルを取得する
     *
     * @param keyValue 設定値
     * @return 表示テキスト
     * @throws IllegalArgumentException このクラスで使用する定数でない場合
     */
    public static final String getLabel(final short keyValue) throws IllegalArgumentException { // NOPMD UnChecked

        switch (keyValue) {

            // 2以上の都道府県の区域等
            case COMPLEX_PREF:
                return COMPLEX_PREF_TEXT;

            // 同一の都道府県の区域内
            case DOUITSU:
                return DOUITSU_TEXT;

            default:
                throw new IllegalArgumentException("Unexpected value: " + keyValue);
        }
    }
}

File: app/seijishikin-jp-normalize/src/components/main/constants/shuushi_doc/v05/shuushiDocV5KatsudouKuikiConstants.ts
export class ShuushiDocV5KatsudouKuikiConstants {

    /** 2以上の都道府県の区域等 */
    static readonly COMPLEX_PREF = 1;

    /** 同一の都道府県の区域内 */
    static readonly DOUITSU = 2;

    /** 2以上の都道府県の区域等 */
    static readonly COMPLEX_PREF_TEXT = "2以上の都道府県の区域等";

    /** 同一の都道府県の区域内 */
    static readonly DOUITSU_TEXT = "同一の都道府県の区域内";

    /**
     * 値に該当するラベルを作成する
     * @param keyValue 値
     * @returns ラベル
     */
    public static getLabel(keyValue: number): string {

        switch (keyValue) {
            case ShuushiDocV5KatsudouKuikiConstants.COMPLEX_PREF:
                return ShuushiDocV5KatsudouKuikiConstants.COMPLEX_PREF_TEXT;

            case ShuushiDocV5KatsudouKuikiConstants.DOUITSU:
                return ShuushiDocV5KatsudouKuikiConstants.DOUITSU_TEXT;

            default:
                return "";
        }
    }
}
